src/object/srv_obj.c

```diff
@@ -1507,13 +1507,21 @@ obj_local_rw_internal(crt_rpc_t *rpc, struct obj_io_context *ioc,
 			goto post;
 		}
 
-		if (ioc->ioc_coc->sc_props.dcp_csum_enabled) {
+		if (ioc->ioc_coc->sc_props.dcp_csum_enabled &&
+		    daos_csummer_initialized(ioc->ioc_coc->sc_csummer)) {
```

Comment:
Since the server uses ULT's, could we simply allocate csummer on the stack and use a copy_inplace version of daos_csummer_copy and thus avoid the allocation?